common/src/com/thoughtworks/go/server/service/ElasticAgentRuntimeInfo.java

```diff
@@ -55,6 +55,11 @@ public void updateSelf(AgentRuntimeInfo newRuntimeInfo) {
         this.elasticPluginId = ((ElasticAgentRuntimeInfo) newRuntimeInfo).getElasticPluginId();
     }
 
+    @Override
+    public boolean isElastic() {
```

Comment:
Hi, @ketan:
Can you expand attributes of AgentRuntimeInfo instead of subclassing it with ElasticAgentRuntimeInfo?  In another PR I am working on cleaning up websocket message json structures. Having type information make serialization and deserialization hard.